doc/docs/Python_Tutorials/Cylindrical_Coordinates.md

```diff
@@ -585,3 +585,202 @@ Note that the volume specified in `get_farfields` via `center` and `size` is in
 Shown below is the far-field energy-density profile around the focal length for both the *r* and *z* coordinate directions for three lens designs with $N$ of 25, 50, and 100. The focus becomes sharper with increasing $N$ due to the enhanced constructive interference of the diffracted beam. As the number of zones $N$ increases, the size of the focal spot (full width at half maximum) at $z = 200$ μm decreases as $1/\sqrt{N}$ (see eq. 17 of the [reference](http://zoneplate.lbl.gov/theory)). This means that doubling the resolution (halving the spot width) requires quadrupling the number of zones.
 
 ![](../images/zone_plate_farfield.png#center)
+
+Nonaxisymmetric Dipole Sources
+-------------------------------
+
+In [Tutorial/Local Density of States/Extraction Efficiency of a Light-Emitting Diode (LED)](Local_Density_of_States.md#extraction-efficiency-of-a-light-emitting-diode-led), the extraction efficiency of an LED was computed using an axisymmetric point-dipole source at $r = 0$. This involved a single simulation with $m = \pm 1$. Simulating a point-dipole source at $r > 0$ (as shown in the schematic below) is more challenging because it is nonaxisymmetric: any point source at $r > 0$ is equivalent to a ring source in cylindrical coordinates. In order to simulate a point-dipole source at $r > 0$, it is necessary to represent the $\delta(r)$ (Dirac delta function) current source as a Fourier-series expansion of $\phi$ (the field angular dependence $\exp(im\phi)$). This computational approach involves two parts: (1) performing a series of simulations for $m = 0, 1, 2, ..., M$ for some cutoff $M$ of the Fourier series (described below), and (2) because of power orthogonality, summing the results from each $m$-simulation in post processing where the $m \neq 0$ terms are multiplied by two to account for the $-m$ solutions.
```

Comment:
What you want is a source that is a 3d delta function:
```
δ(x)δ(y)δ(z) = δ(r)δ(φ)δ(z) / 2πr
```
and then to frame this in terms of axisymmetric simulations with exp(imφ) dependence you need to expand δ(φ) in a Fourier series
```
\delta(\phi) = \frac{1}{2\pi} \sum_m e^{im\phi}
```
up to a normalization.

Reply:
Regarding power orthogonality:

* Physically, the total field E(x,y,z) is a sum of E_m(r,z)e^{im\phi} terms, one for the solution at each m (similarly for H).
* When you compute things like the total Poynting flux, however, you are integrating \Re [ E \times H^*] over a surface that includes an integral over φ.  In doing so, you'll see that all of the cross terms of different m's in E \times H^* integrate to zero (thanks to Fourier orthogonality), so you end up just getting a sum over the Poynting fluxes calculated separately for each m.   In short, the total power is the sum of the powers for each m.

Reply:
Consider also the case of an x-polarized source, i.e. J proportional to `x̂`.  Then you have:
```
δ(x)δ(y)δ(z) x̂ = δ(r)δ(φ)δ(z) r̂ / 2πr
```
where the key point is that even though `r̂` is φ-dependent, because of the `δ(φ)` you are only evaluating `r̂` at φ=0 where it is the *same* as `x̂`.

This is different from point an `x̂` source at r=0, because then you write it with no `δ(φ)` as `x̂ = r̂ cos(ϕ) - ϕ̂ sin(ϕ)` and the cos and sin terms are why you do the calculation at m=1.

Reply:
I have expanded the discussion to address all of these points.